### DISCUSSION TITLE
How to use bullmq in a micro service

### DISCUSSION THREAD

**Discussion by 850036165** [taskforcesh/bullmq#2652] [Q&A] [answered]
In an asynchronous task processing queue, a job needs to request the interface of another service, wait for feedback on the completion of another interface call task, and update the task status. 
How to 'suspend' the job and then return the result to decide whether it is success or failed?
Thanks

```
const worker = new Worker(queueName, async (job: Job) => {
 const res =  await axios.post('algorithm/process',{data:job.data})  //request another system
// suspend and waiting for the result
});

 // after processing finished
@Controller()
export class AppController {
  @Post('algorithm/callback')
  async handleCallback(@Body() result: any) {
    //update job status with response
  }
}
```

**Comment by manast** [comment, marked as answer]
In javascript you cannot do that, when you have an async function it must complete or fail (or run forever if it does not timeout or something). If you control the service "algorithm/process" what you typically do is that you post some job there, and then return immediately, when the algorithm/process completes it posts the result in another queue, a result queue, that some other service listens to, and so on. Maybe this blog post can give you some ideas: https://blog.taskforce.sh/do-not-wait-for-your-jobs-to-complete/

**Reply by 850036165** [reply]
Very helpful, thanks!
